test/e2e/aro.go

```diff
@@ -0,0 +1,69 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+)
+
+const (
+	wSsh = "99-worker-ssh"
+	mSsh = "99-master-ssh"
+)
+
+// The high level idea of these tests is to definitevily say that a cluster is an ARO cluster.
+var _ = Describe("Get ARO cluster", func() {
+	It("should be possible to definitely confirm cluster is ARO", func() {
+		ctx := context.Background()
+
+		// Get cluster object
+		co, err := clients.AROClusters.AroV1alpha1().Clusters().Get(context.Background(), "cluster", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		mcp, err := clients.MachineConfig.MachineconfigurationV1().MachineConfigPools().List(ctx, metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Expect 99-worker-ssh and 99-master-ssh to exist
+		masterSsh, workerSsh := false, false
+		for _, m := range mcp.Items {
+			for _, mc := range m.Spec.Configuration.Source {
+				if mc.Name == wSsh {
+					workerSsh = true
+				}
+				if mc.Name == mSsh {
+					masterSsh = true
+				}
+			}
+		}
+		Expect(workerSsh).To(BeTrue())
+		Expect(masterSsh).To(BeTrue())
+
+		// Expect ACR Domain
+		Expect(co.Spec.ACRDomain).To(BeEquivalentTo("arointsvc.azurecr.io"))
```

Comment:
This will only pass in CI. During rp releases we create prod clusters and this would have the prod ACR value

Reply:
just updated to cover these cases along with the suggestions you made above. Please rereview at your convenience